server/controllers/contest.js

```diff
@@ -60,3 +62,45 @@ exports.getContests = asyncHandler(async (req, res, next) => {
     throw new Error('failed to get contests');
   }
 });
+
+// handler for create submission by contest id
+exports.createSubmissionByContestId = asyncHandler(async (req, res, next) => {
+  const contestId = req.params.id;
+  const s3UrlArray = req.body.data;
+  const userId = req.user.id;
+  if (!ObjectId.isValid(contestId)) {
+    return res.status(400).json({
+      error: 'Contest ID is invalid.',
+    });
+  }
+  try {
+    const userExist = await Submission.findOne({ userId: userId, contestId: contestId });
+    if (userExist) {
+      const submissionData = await Submission.findByIdAndUpdate(
+        userExist._id,
+        { files: s3UrlArray },
```

Comment:
Won't updating the files array replace it? (We will loose the previous URLs)

Reply:
I was thinking to limit up to 5 designs for each user, but I can change append all URLs.